apps/ios/Shared/Model/BGManager.swift

```diff
@@ -34,35 +39,49 @@ class BGManager {
         }
     }
 
-    func schedule() {
+    func schedule(soon: Bool = false) {
         if !ChatModel.shared.ntfEnableLocal {
             logger.debug("BGManager.schedule: disabled")
             return
         }
         logger.debug("BGManager.schedule")
         let request = BGAppRefreshTaskRequest(identifier: receiveTaskId)
-        request.earliestBeginDate = Date(timeIntervalSinceNow: bgRefreshInterval)
+        request.earliestBeginDate = Date(timeIntervalSinceNow: soon ? bgRefreshInterval : runInterval)
         do {
             try BGTaskScheduler.shared.submit(request)
         } catch {
             logger.error("BGManager.schedule error: \(error.localizedDescription)")
         }
     }
 
+    var runInterval: TimeInterval {
+        switch ChatModel.shared.notificationMode {
+        case .instant: maxBgRefreshInterval
+        case .periodic: periodicBgRefreshInterval
+        case .off: bgRefreshInterval
+        }
+    }
+
+    var shouldRun: Bool {
```

Comment:
```suggestion
    var lastRanLongAgo: Bool {
```

`shouldRun` isn't very descriptive name